enginetest/queries/script_queries.go

```diff
@@ -138,6 +138,125 @@ var ScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "intersection tests",
+		SetUpScript: []string{
+			"create table a (m int, n int);",
+			"insert into a values (1,2), (2,3), (3,4);",
+			"create table b (m int, n int);",
+			"insert into b values (1,2), (1,3), (3,4);",
+			"create table c (m int, n int);",
+			"insert into c values (1,3), (1,3), (3,4);",
+			"create table t1 (i int);",
+			"insert into t1 values (1), (2), (3);",
+			"create table t2 (i float);",
+			"insert into t2 values (1.0), (1.99), (3.0);",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "table a intersect table b order by m, n;",
+				Expected: []sql.Row{
+					{1, 2},
+					{3, 4},
+				},
+			},
+			{
+				Query: "table a intersect table c order by m, n;",
+				Expected: []sql.Row{
+					{3, 4},
+				},
+			},
+			{
+				Query: "table c intersect distinct table c order by m, n;",
+				Expected: []sql.Row{
+					{1, 3},
+					{3, 4},
+				},
+			},
+			{
+				Query: "table c intersect all table c order by m, n;",
+				Expected: []sql.Row{
+					{1, 3},
+					{1, 3},
+					{3, 4},
+				},
+			},
+			{
+				// Resulting type is string for some reason
+				Skip:  true,
+				Query: "table t1 intersect table t2;",
+				Expected: []sql.Row{
+					{1},
+					{3},
+				},
+			},
+			{
+				// Field indexing error
+				Skip:  true,
+				Query: "table t1 intersect table t2 order by i;",
```

Comment:
I'm seeing that Union is now `SetArithemticOp`, so I'm less certain about the cause now. I thought we were trying to index `order by i` with `[i,i]`. We pick the closest match from the right. But `GetField(1)` is invalid, and will error. We should fix either way before shipping

Reply:
Similar to `INTERSECT` and `EXCEPT`, `UNION` along with `ORDER BY` also fail in the same way. I suspect it has something to do with being unable to find an exact column match since the types are different. 

Will continue investigating